test/test.js

```diff
@@ -131,8 +136,114 @@ describe('Service', function() {
       assert.notExists(properties.getProperty(key));
     });
   });
+
+  describe('#hasAccess()', function() {
+    it('should use the lock to prevent concurrent access', function(done) {
+      var token = {
+        granted_time: 100,
+        expires_in: 100,
+        refresh_token: 'bar'
+      };
+      var properties = new MockProperties();
```

Comment:
Rather than declaring:
```js
var properties = new MockProperties();
properties.setProperty(...)
```
twice, can you allow `MockProperties` to accept default properties? It would make sense and is small [2 line change](https://github.com/googlesamples/apps-script-oauth2/blob/master/test/mocks/properties.js).

```js
var MockProperties = function(store) {
  this.store = store || {};
  this.counter = 0;
};
```

then create `MockProperties`.

```js
var properties = new MockProperties({'oauth2.test': JSON.stringify(token)});
```

Reply:
Done. Switched to using it in most cases, but not all.